### PR TITLE
Signup form settings

### PR DESCRIPTION
**Pull request by umida-boltaeva**
I made the React side of the signup page. CSS side has not finished yet.

### CODE REVIEW

File: src/signup/signup.js [hunk 1]
```diff
@@ -1,10 +1,75 @@
...
+import CssBaseline from '@material-ui/core/CssBaseline';
+import Typography from '@material-ui/core/Typography';
+import Button from '@material-ui/core/Button';
+import styles from "./styles"
+import { Form } from "reactstrap";
+const firebase = require("firebase");
 
 class SignupComponent extends React.Component {
     
     render() {
```
Comment (nickmcmillan) [inline src/signup/signup.js]: It's normally a convention to put render() as the last method in a React class.

File: src/signup/signup.js [hunk 2]
```diff
@@ -1,10 +1,75 @@
...
-        return(<div>Hello World from Signup</div>)
+
+        const { classes } = this.props;
```
Comment (nickmcmillan) [inline src/signup/signup.js]: What's your intention here? This makes me assume that the parent component will pass all the styling classes down through props. Therefore the parent component will control all of the styling for this component... That might make this component less portable and self-contained. But I'm curious to see if there are advantages to doing it that I'm not aware of.

**Review by nickmcmillan — APPROVED** [review]
This is really interesting stuff. I didn't know you planned on having user-registration functionality - but I suppose if that functionality comes easily with Firebase then why not?

File: src/signup/signup.js (hunk 1 above)
Comment (umida-boltaeva) [reply src/signup/signup.js]: Yep, I corrected it following the convention.

File: src/signup/signup.js (hunk 2 above)
Comment (umida-boltaeva) [reply src/signup/signup.js]: Well, here classes property is not passed by the parent component. I'm using `withStyles` from Material-UI. You can see it at the end of the code: `export default withStyles(styles)(SignupComponent)`.  It injects classes to the props:
https://material-ui.com/guides/typescript/#usage-of-withstyles

File: src/signup/signup.js (hunk 1 above)
Comment (umida-boltaeva) [reply src/signup/signup.js]: Yep, updated.

**Pull request merged**